content/en/blog/2019/alb-ingress-gateway-iks/index.md

```diff
@@ -0,0 +1,389 @@
+    {{< /text >}}
+
+1.  Extract the certificate and the key from the secret provided for the ingress gateway:
+
+    {{< text bash >}}
+    $ mkdir ingress_gateway_certs
+    $ kubectl get secret $INGRESS_GATEWAY_SECRET --namespace=default -o yaml | grep 'tls.key:' | cut -f2 -d: | base64 --decode > ingress_gateway_certs/tls.key
+    $ kubectl get secret $INGRESS_GATEWAY_SECRET --namespace=default -o yaml | grep 'tls.crt:' | cut -f2 -d: | base64 --decode > ingress_gateway_certs/tls.crt
+    $ ls -al ingress_gateway_certs
+    -rw-r--r--   1 user  staff  1679 Sep 11 07:55 tls.key
+    -rw-r--r--   1 user  staff  3921 Sep 11 07:55 trusted.crt
+    {{< /text >}}
+
+1.  Extract the certificate and the key from the secret provided for ALB:
+
+    {{< text bash >}}
+    $ mkdir alb_certs
+    $ kubectl get secret $ALB_SECRET --namespace=default -o yaml | grep 'tls.key:' | cut -f2 -d: | base64 --decode > alb_certs/client.key
+    $ kubectl get secret $ALB_SECRET --namespace=default -o yaml | grep 'tls.crt:' | cut -f2 -d: | base64 --decode > alb_certs/client.crt
+    $ ls -al alb_certs
+    -rw-r--r--   1 user  staff  3738 Sep 11 07:57 client.crt
+    -rw-r--r--   1 user  staff  1675 Sep 11 07:57 client.key
+    {{< /text >}}
+
+1.  Download the issuer certificate of the [Let's Encrypt](https://letsencrypt.org) certificate, which is the
+    issuer of the certificates provided by IKS.
+
+    {{< text bash >}}
+    $ curl https://letsencrypt.org/certs/trustid-x3-root.pem --output trustid-x3-root.pem
+    {{< /text >}}
+
+1.  Append the issuer certificate of [Let's Encrypt](https://letsencrypt.org) to the certificate of ingress gateway
+    (currently required for ALB):
+
+    {{< text bash >}}
+    $ cat ingress_gateway_certs/tls.crt trustid-x3-root.pem > trusted.crt
+    {{< /text >}}
+
+1.  Create Kubernetes secrets to be used by Istio ingress gateway and ALB to establish mutual TLS between them. Note
+    that the name of the secrets for the Istio ingress gateway must be exactly as in the commands.
+
+    {{< warning >}}
+    The certificates provided by IKS expire every 90 days and are automatically renewed by
+    IKS 37 days before they expire.
+    You will have to recreate the secrets by rerunning the instructions of this section every time the secrets provided
+    by IKS are updated. You may want to use scripts or operators to automate this and keep the
+    secrets in sync.
+    {{< /warning >}}
+
+    {{< text bash >}}
+    $ kubectl create -n istio-system secret tls istio-ingressgateway-certs --key ingress_gateway_certs/tls.key --cert trusted.crt
+    $ kubectl create -n istio-system secret generic istio-ingressgateway-ca-certs --from-file=trustid-x3-root.pem
+    $ kubectl create secret generic alb-certs -n istio-system --from-file=trusted.crt --from-file=alb_certs/client.crt --from-file=alb_certs/client.key
+    secret "istio-ingressgateway-certs" created
+    secret "istio-ingressgateway-ca-certs" created
+    secret "alb-certs" created
```

Comment:
maybe a follow up blog - would be good to see steps with SDS gateway.

Reply:
I can write a follow-up blog, but it is a pretty simple transformation from file-mounted certificates to the SDS provided one.

Reply:
I think it would be good, since there are a lot of interest with SDS gateway.